utils/gen_config.py

```diff
@@ -0,0 +1,174 @@
+# coding=utf-8
+"""Update scan facts in config file given scan data.
+
+Run this script as a command line tool to use csv with data about hosts and
+profiles and report file from rho scan to generate camayoc config file.
+
+.. note::
+
+    This tool reads in your camayoc config file in $XDG_CONFIG_HOME and outputs
+    it to the $CWD with new data. It discards old hosts data but uses auth,
+    profile, and any other data provided in your $XDG_CONFIG_HOME config file
+    (if it exists). This does not modify your $XDG_CONFIG_HOME config file
+    directly, that is left to you. Always backup your old config file before
+    you replace it with the generated one!
+
+Example:
+    .. code-block:: bash
+
+        python gen_config.py --profilecsv profiles.csv --reportcsv report.csv
+"""
+from camayoc import config
+import uuid
+from camayoc.exceptions import ConfigFileNotFoundError
+import os
+import csv
+import yaml
+import argparse
+
+
+base_config = """
+rho:
+  auths:
+    - username: root
+      name: default
+      sshkeyfile: $HOME/.ssh/id_rsa
+"""
+
+
+def get_config():
+    """Gather existing config file or create one.
+
+    If a config file is found by camayoc, use that.
+    Otherwise, create the file and directory if necessary.
+    """
+    try:
+        cfg = config.get_config()
+    except ConfigFileNotFoundError:
+        cfg = yaml.load(base_config)
+
+    return cfg, os.path.join(os.getcwd(), 'config.yaml')
+
+
+def gen_hosts(profilefile, reportfile):
+    """Generate list of hosts from profile and report data.
+
+    Args:
+        profilefile (string path to file):
+            Expects csv with columns:
+                - name
+                - ip
+                - profile
+                - auth
+        reportfile (string path to file):
+            Expects csv from rho scan
+
+    .. note::
+        It will generated all new hosts, old host data is discarded.
+    """
+    hosts = []
+    with open(profilefile) as csvfile:
+        hostdata = csv.DictReader(csvfile)
+        for row in hostdata:
+            hosts.append({
+                'hostname': row.get('name', ''),
+                'ip': row.get('ip', ''),
+                'provider': 'vcenter' if 'vc' in row.get('name', '') else '',
+                'facts': {},
+                'profile': row.get('profile', uuid.uuid4()),
+                'auth': row.get('auth', ''),
+            })
+    with open(reportfile) as csvfile:
+        scan_results = csv.DictReader(csvfile)
+        for row in scan_results:
+            for host in hosts:
+                if host['ip'] == row.get('connection.host', ''):
```

Comment:
Not a blocker but since you are using `==` you don't need to care if there is no `connection.host` and accept the `None` returned by get when the value is not available:

```python
>>> 'a' == None
False
```